spec/rack/i18n_locale_switcher_spec.rb

```diff
@@ -294,22 +307,48 @@
 
       it_should_behave_like "a redirect with the default locale"
     end
-    
+
     context "exceptions" do
 
       let :options do
         { :redirect => :path, :except => /^\/assets/ }
       end
-      
+
       it "should not redirect if the path is exempt" do
-        [ "http://example.com/assets",   
-          "http://de.example.com/assets/foo/bar",
-          "http://example.com/assets/"
+        ["http://example.com/assets",
+         "http://de.example.com/assets/foo/bar",
+         "http://example.com/assets/"
         ].each do |url|
           get url
           last_response.should_not be_redirect
         end
       end
     end
   end
-end
\ No newline at end of file
+
+  context "save_to_cookie" do
+    let(:options) do
+      { save_to_cookie: true, cookie: 'cookie_name_for_locale' }
+    end
+
+    [
+      ["path", "http://example.com/es"],
+      ["param", "http://example.com", { 'locale' => 'es' }, {}],
+      ["host", "http://es.example.com"],
+      ["headers", "http://example.com", {}, { "HTTP_ACCEPT_LANGUAGE" => "es" }],
+      ["cookie", "http://example.com", {}, { "HTTP_COOKIE" => "cookie_name_for_locale=es" }],
+    ].each do |src, url, params, env|
+      it "should set locale in the cookie header when reading from #{src}" do
+        if params || env
```

Comment:
Wouldn't it be ok to always send `get url, params, env` even if `params` and/or `env` are `nil`?
Conditionals in tests makes it hard to understand the tests and increase the chances of bugs in the tests (which don't have tests themselves).

Reply:
yeah changing it to remove this fork